docs/src/main/asciidoc/amqp-guide.adoc

```diff
@@ -0,0 +1,312 @@
+
+The `channel-name` segment must match the value set in the `@Incoming` and `@Outgoing` annotation:
+* `generated-price` -> sink in which we write the prices
+* `prices` -> source in which we read the prices
+
+[source]
+----
+# Configures the AMQP broker credentials.
+amqp-username=quarkus
+amqp-password=quarkus
+
+# Configure the AMQP connector to write to the `prices`  address
+mp.messaging.outgoing.generated-price.connector=smallrye-amqp
+mp.messaging.outgoing.generated-price.address=prices
+mp.messaging.outgoing.generated-price.durable=true
+
+# Configure the AMQP connector to read from the `prices` queue
+mp.messaging.incoming.prices.connector=smallrye-amqp
+mp.messaging.incoming.prices.durable=true
+----
+
+More details about this configuration is available on the https://smallrye.io/smallrye-reactive-messaging/#_interacting_using_amqp[SmallRye Reactive Messaging AMQP connector] documentation.
+
+NOTE: What about `my-data-stream`? This is an in-memory stream, not connected to a message broker.
+
+== The HTML page
+
+Final touch, the HTML page reading the converted prices using SSE.
+
+Create the `src/main/resources/META-INF/resources/prices.html` file, with the following content:
+
+[source, html]
+----
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>Prices</title>
+
+    <link rel="stylesheet" type="text/css"
+          href="https://cdnjs.cloudflare.com/ajax/libs/patternfly/3.24.0/css/patternfly.min.css">
+    <link rel="stylesheet" type="text/css"
+          href="https://cdnjs.cloudflare.com/ajax/libs/patternfly/3.24.0/css/patternfly-additions.min.css">
+</head>
+<body>
+<div class="container">
+
+    <h2>Last price</h2>
+    <div class="row">
+    <p class="col-md-12">The last price is <strong><span id="content">N/A</span>&nbsp;&euro;</strong>.</p>
+    </div>
+</div>
+</body>
+<script src="https://code.jquery.com/jquery-3.3.1.min.js"></script>
+<script>
+    var source = new EventSource("/prices/stream");
+    source.onmessage = function (event) {
+        document.getElementById("content").innerHTML = event.data;
+    };
+</script>
+</html>
+----
+
+Nothing spectacular here. On each received price, it updates the page.
+
+== Get it running
+
+If you followed the instructions, you should have AMQP broker running.
+Then, you just need to run the application using:
+
+[source, shell]
+----
+mvn compile quarkus:dev
+----
+
+Open `http://localhost:8080/prices.html` in your browser.
+
+NOTE: If you started the AMQP broker with docker compose, stop it using `CTRL+C` followed by `docker-compose down`.
+
+== Running Native
+
+You can build the native executable with:
+
+[source, shell]
+----
+mvn package -Pnative
+----
+
+== Going further
+
+This guide has shown how you can interact with AMQP using Quarkus.
+It utilizes MicroProfile Reactive Messaging to build data streaming applications.
+
+If you did the Kafka quickstart, you have realized that it's the same code.
```

Comment:
I did realize that as I reviewed the other one :).

Reply:
which means that all your suggestions are also valid for the Kafka guide ;-)